daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -122,6 +132,25 @@ private[lf] final case class Compiler(packages: PackageId PartialFunction Packag
   /** Environment mapping names into stack positions */
   private var env = Env()
 
+  private val withLabel: (AnyRef, SExpr) => SExpr =
+    profiling match {
+      case NoProfile => { (_, expr) =>
+        expr
+      }
+      case FullProfile => { (label, expr) =>
+        expr match {
+          case SELabelClosure(_, expr1) => SELabelClosure(label, expr1)
+          case _ => SELabelClosure(label, expr)
+        }
+      }
```

Comment:
`{}` are not really part of the function syntax, nor the `case` syntax, irrespective of IDEA's obsession with them. There are occasionally good reasons to use them with functions, but never with `case` per se.

```suggestion
      case NoProfile => (_, expr) => expr
      case FullProfile => (label, expr) =>
        expr match {
          case SELabelClosure(_, expr1) => SELabelClosure(label, expr1)
          case _ => SELabelClosure(label, expr)
        }
```

Reply:
I find the `case x => y => z` syntax quite confusing because the two occurrences of `=>` don't mean exactly the same thing. I was looking for some scoping construct to put in between them. Could I just use `(...)`?

Reply:
> Could I just use `(...)`?

Yes.